src/paint.js

```diff
@@ -598,19 +608,35 @@ function setupPaint({ $, qlik }) {
               return colors[i];
             })
             .style("opacity", tileOpacity)
-            .on("mouseenter", function (d, i) {
-              d3.selectAll('[fill="' + colors[i] + '"]')
+            .on("mouseenter", function (dataObject, index) {
+              d3.selectAll('[fill="' + colors[index] + '"]')
                 .classed({
                   "bordered": false,
                   "borderedHover": true
                 });
             })
-            .on("mouseleave", function (d, i) {
-              d3.selectAll('[fill="' + colors[i] + '"]')
+            .on("mouseleave", function (dataObject, index) {
+              d3.selectAll('[fill="' + colors[index] + '"]')
                 .classed({
                   "bordered": true,
                   "borderedHover": false
                 });
+            })
+            .on('click', function (d , i){
+              var rectsArray = d3.selectAll('[fill="' + colors[i] + '"]');
```

Comment:
Please rename `rectsArray` to something that indicates that we are dealing with the svg rectangles that we selected.
Also we need a better way to handle selection, as we are also including the `html` tag. For example if there are no hits, `rectsArray` first element will become the `html` tag. If we have some rectangles to select, then the `html` tag will always be the last. We don't want to include the `html` tag.

Reply:
The `svgRectsArray` has the parent Node